### PR TITLE
✨ Add target_window to actions

### PR DESCRIPTION
**Pull request by RocKhalil**
Added the ability to change the target window of the action.
For example, I want to open an `Import` page in a different window, I'll need to do:

`target_window :_blank`

### CODE REVIEW

**Comment by RocKhalil** [comment]
@jdufresne @mshibuya not sure why I'm marked as a first time contributor, but anyway, can you guys check this please ?

File: lib/rails_admin/config/actions/base.rb [hunk 1]
```diff
@@ -78,6 +78,11 @@ class Base
           true
         end
 
+        # Target window [_self, _blank]
+        register_instance_option :target_window do
+          :_self
```
Comment (jdufresne) [inline lib/rails_admin/config/actions/base.rb:83]: Could we default to no attribute instead? Maybe that would be handled by `nil`? In the normal cause I wouldn't expect `target="_self"`, just nothing.

**Review by jdufresne — COMMENTED** [review]
would it be possible to add a test showing the rendered result?

File: lib/rails_admin/config/actions/base.rb (hunk 1 above)
Comment (RocKhalil) [reply lib/rails_admin/config/actions/base.rb]: Removed the default value, however, I'm not sure what's the test that can cover this change;
Any idea ?

Comment (jdufresne) [reply lib/rails_admin/config/actions/base.rb]: >  Any idea ?

Would id be possible to write a test that renders HTML with this option? This HTML either contains the attribute `target="_blank"` or it doesn't.

**Comment by RocKhalil** [comment]
@jdufresne I've added a test, however, I'm not able to install `mysql` gem on my mac for some reason, so I wasn't able to run it to validate it.
Can you please double check ?

File: lib/rails_admin/config/actions/base.rb [hunk 2]
```diff
@@ -78,6 +78,11 @@ class Base
           true
         end
 
+        # Target window [_self, _blank]
+        register_instance_option :target_window do
```
Comment (mshibuya) [inline lib/rails_admin/config/actions/base.rb:82]: To me, `link_target` feels like to be a better name. What do you think?

File: lib/rails_admin/config/actions/base.rb [hunk 3]
```diff
@@ -78,6 +78,11 @@ class Base
           true
```
Comment (mshibuya) [inline lib/rails_admin/config/actions/base.rb:78]:
```suggestion
          target_window.blank?
```
Or maybe `target_window != :_blank`

File: app/helpers/rails_admin/application_helper.rb [hunk 4]
```diff
@@ -163,7 +163,14 @@ def menu_for(parent, abstract_model = nil, object = nil, only_icon = false)
...
               end
-            content_tag(:a, label, {href: href}.merge(action.pjax? ? {class: ['pjax']} : {}))
+
+            extra_classes = []
+            extra_classes << 'pjax' if action.pjax? && action.target_window != :_blank
+
+            extra_tags = {}
+            extra_tags[:target] = action.target_window if action.target_window.present?
+
+            content_tag(:a, label, {href: href, class: extra_classes}.merge(extra_tags))
```
Comment (mshibuya) [inline app/helpers/rails_admin/application_helper.rb:173]:
```suggestion
            content_tag(:a, label, {href: href, target: action.target_window}.merge(action.pjax? ? {class: ['pjax']} : {}))
```

Comment (RocKhalil) [reply app/helpers/rails_admin/application_helper.rb:173]: One of the issues of PJAX, is that, when enabled, it was removing the effect of the `target=_blank` for some reason

File: lib/rails_admin/config/actions/base.rb (hunk 2 above)
Comment (RocKhalil) [reply lib/rails_admin/config/actions/base.rb:82]: Yes, makes sense! why not

**Comment by mshibuya** [comment]
Merged in, thanks for the contribution 🙌

**Comment by eliten00b** [comment]
Can we have this back ported and release on version 2.x?

**Pull request merged**